executor/cook/progress.py

```diff
@@ -138,6 +156,10 @@ def tail(self, sleep_time_ms):
         try:
             sleep_param = sleep_time_ms / 1000
 
+            if os.path.exists(self.target_file) and not os.path.isfile(self.target_file):
```

Comment:
Do we need this?  Why wouldn't we change for `!isfile` after the file is created in the loop below?

Reply:
I think the two `if` clauses address different scenarios. 
- This `if` specifies that the location specified already exists and is not a file.

The `while` loop will keep running for inputs like `/dev/null` which aren`t files and which will never be created. We would only stop looping once `task_completed_signal` is set. The `if` condition below is `true` only after the task completed and is a notification that we were expecting to read from a file which was never created.

Reply:
A "non-file" may be created after this `if` statement.  I may not be following.

Reply:
Let's talk offline